ossec_security/assets/logs/ossec-security.yaml

```diff
@@ -0,0 +1,708 @@
+              data deleted"
+          grok:
+            supportRules: |-
+              DATETIME (%{date("MMM  d HH:mm:ss"):}|%{date("MMM d HH:mm:ss"):})
+              SKIPHEADER %{regex("[^:]*"):}
+              TILL_SINGLE_QUOTES %{regex("[^']*"):}
+            matchRules: >-
+              user_parsing_rule %{DATETIME} %{SKIPHEADER}: (new user:|new
+              account added -|account deleted -) %{data:user:keyvalue("=","/")}
+
+
+              group_parsing_rule %{DATETIME} %{SKIPHEADER}: new group: %{data:group:keyvalue("=","/")}
+
+
+              add_user_failed_rule %{DATETIME} %{SKIPHEADER}: failed adding user '%{regex("[a-z0-9-_]*"):usr.name}', data deleted
+
+
+              user_info_changed_rule %{DATETIME} %{SKIPHEADER}: changed user '%{TILL_SINGLE_QUOTES:usr.name}' information
+        - type: attribute-remapper
+          name: Map `user.account`, `user.name` to `usr.name`
+          enabled: true
+          sources:
+            - user.account
+            - user.name
+          sourceType: attribute
+          target: usr.name
+          targetType: attribute
+          preserveSource: false
+          overrideOnConflict: false
+        - type: attribute-remapper
+          name: Map `user.UID`, `user.uid` to `usr.id`
+          enabled: true
+          sources:
+            - user.UID
+            - user.uid
+          sourceType: attribute
+          target: usr.id
+          targetType: attribute
+          preserveSource: false
+          overrideOnConflict: false
+    - type: pipeline
+      name: Processing of FTPD Logs
+      enabled: true
+      filter:
+        query: "@rule_id:[11100 TO 11113] "
+      processors:
+        - type: grok-parser
+          name: Parsing of FTPD logs.
+          enabled: true
+          source: log
+          samples:
+            - "May 28 15:52:51 ub20 ftpd[27654]: User dummy timed out after 910
+              seconds at Sun May 28 15:52:51 2024"
+            - "May 28 15:50:36 ub20 ftpd[28370]: FTP LOGIN FROM 10.10.10.10,
+              dummy"
+            - "May 29 11:04:16 ub20 ftpd[417946]: connect from 10.10.10.10"
+            - >
+              May 28 15:14:02 ub20 ftpd[28616]: FTPD: EXPORT file local
+              %APPDATA%\Microsoft\Windows\Start Menu\Programs\Startup, remote
+              sample.html
+            - "May 28 15:50:36 ub20 ftpd[28370]: connection from 10.10.10.10 at
+              Sun May 28 15:50:36 2006"
+          grok:
+            supportRules: |-
+              DATETIME (%{date("MMM  d HH:mm:ss"):}|%{date("MMM d HH:mm:ss"):})
+              SKIPHEADER %{regex("[^:]*"):}
+            matchRules: >-
+              ftpd_file_transfer_rule %{DATETIME} %{SKIPHEADER}: %{notSpace}:
+              %{word:action} file local (%{data:local_file_path})?, remote(
+              %{data:remote_file_path:nullIf("")})?
+
+
+              user_timeout_rule %{DATETIME} %{SKIPHEADER}: User %{notSpace:usr.name} timed out after %{integer:timeout} seconds at %{data}
```

Comment:
you can replace the use of %{data} to match within messages using %{date("EEE MMM dd HH:mm:ss yyyy")} instead
 we try to reduce the usage of %{data} because it can be expensive for long message. It might not be your case but its a precaution in case you are unexpected large logs, it will prevent breaking logs for other customers

Reply:
We have used %{data} since we were not sure about datetime format. There is a possibility it might change as per the system configuration. Hence we have used %{data} to keep parsing rule generalized.